crates/data_components/src/databricks/sql_warehouse/datatypes.rs

```diff
@@ -0,0 +1,455 @@
+            }
+            Some(Ok(Token::SmallInt)) => {
+                self.advance();
+                Ok(ArrowDataType::Int16)
+            }
+            Some(Ok(Token::String | Token::Variant)) => {
+                self.advance();
+                Ok(ArrowDataType::Utf8)
+            }
+            Some(Ok(Token::Timestamp)) => {
+                self.advance();
+                Ok(ArrowDataType::Timestamp(
+                    TimeUnit::Nanosecond,
+                    Some("UTC".into()),
+                ))
+            }
+            Some(Ok(Token::TimestampNtz)) => {
+                self.advance();
+                Ok(ArrowDataType::Timestamp(TimeUnit::Nanosecond, None))
+            }
+            Some(Ok(Token::TinyInt)) => {
+                self.advance();
+                Ok(ArrowDataType::Int8)
+            }
+            Some(Ok(Token::Array)) => {
+                self.advance();
+                self.expect(&Token::LAngle)?;
+                let inner_type = self.parse_data_type()?;
+                self.expect(&Token::RAngle)?;
+                let field = ArrowField::new("item", inner_type, true);
+                Ok(ArrowDataType::List(Arc::new(field)))
+            }
+            Some(Ok(Token::Map)) => self.parse_map(),
+            Some(Ok(Token::Struct)) => self.parse_struct(),
+            Some(Ok(Token::Object)) => self.parse_object(),
+            _ => Err(format!("Unexpected token: {:?}", self.current)),
+        }
+    }
+
+    fn parse_map(&mut self) -> Result<ArrowDataType, String> {
+        self.advance();
+        self.expect(&Token::LAngle)?;
+        let key_type = self.parse_data_type()?;
+        self.expect(&Token::Comma)?;
+        let value_type = self.parse_data_type()?;
+        self.expect(&Token::RAngle)?;
+        let key_field = Arc::new(ArrowField::new("key", key_type, false));
+        let value_field = Arc::new(ArrowField::new("value", value_type, true));
+        let entry_struct = Arc::new(ArrowField::new_struct(
+            "entries",
+            vec![key_field, value_field],
+            true,
+        ));
+        Ok(ArrowDataType::Map(entry_struct, false))
+    }
+
+    fn parse_struct(&mut self) -> Result<ArrowDataType, String> {
+        self.advance();
+        self.expect(&Token::LAngle)?;
+        let mut fields = Vec::new();
+        if self.current != Some(Ok(Token::RAngle)) {
+            loop {
+                let field = self.parse_field()?;
+                fields.push(field);
+                if self.current == Some(Ok(Token::Comma)) {
+                    self.advance();
+                    if self.current == Some(Ok(Token::RAngle)) {
+                        break;
+                    }
+                } else {
+                    break;
+                }
+            }
+        }
+        self.expect(&Token::RAngle)?;
+        Ok(ArrowDataType::Struct(fields.into()))
+    }
+
+    fn parse_object(&mut self) -> Result<ArrowDataType, String> {
```

Comment:
This looks the same as parse_struct?

Reply:
It is duplicated code. You can cast an object into a struct. I'm removing it because I didn't realize in the docs that it said Objects cannot be stored in table columns and must be cast so I'm not sure it's useful having this code around. https://docs.databricks.com/aws/en/sql/language-manual/data-types/object-type